core/sdk-core/src/main/java/software/amazon/awssdk/core/client/builder/SdkDefaultClientBuilder.java

```diff
@@ -226,6 +230,7 @@ private SdkClientConfiguration finalizeAsyncConfiguration(SdkClientConfiguration
      */
     private SdkClientConfiguration finalizeConfiguration(SdkClientConfiguration config) {
         return config.toBuilder()
+                     .option(TIMEOUT_EXECUTOR_SERVICE, resolveTimeoutExecutorService())
```

Comment:
Initial thought without looking further in the code first: We already have an internal scheduled executor service for retries. Can we use the same one for timeouts?

Reply:
yeah, we can, probably need to rename the existing`ASYNC_RETRY_EXECUTOR_SERVICE` to `SDK_INTERNAL_EXECUTOR_SERVICE`(or something else)?